### PR TITLE
G2P tests and schema update

### PR DESCRIPTION
**Pull request by david4096**
This PR is for updating g2p against changes to the schemas and stubbing out some tests.

`test_g2p` has been moved from unit to end_to_end tests since it relies on test data and some unit tests have been added.

Bugfix to namespace splitting. It wasn't parsing URLs that were wrapped in `<url>` correctly.

Removed some unused code. Includes recent changes to master.

(WIP) to bring in line with https://github.com/ga4gh/schemas/pull/560

### CODE REVIEW

**Comment by david4096** [comment]
The server code has been refactored to match with the schema changes and there is some scaffolding for putting in the remaining tests. This doesn't provide any new functionality but adds a SimulatedG2PDataset for returning mock associations.

**Comment by jeromekelleher** [comment]
We've pulled in a bunch of extra commits from master here @david4096, which is making the changes hard to see. Can you make a separate update PR to pull the changes from master into g2p, and then rebase this PR off upstream/g2p please?

File: ga4gh/datamodel/genotype_phenotype.py [hunk 1]
```diff
@@ -9,15 +9,55 @@
...
+        fpa = self.toProtocolElement()
+        fpas = []
+        if location or drug or disease:
+            fpas.append(fpa)
+        # TODO this seems like a hack
+
+        def toProtocolElement(self):
+            return self
+
+        for fpa in fpas:
```
Comment (jeromekelleher) [inline ga4gh/datamodel/genotype_phenotype.py]: I'm confused here -- what's this doing?

**Comment by jeromekelleher** [comment]
This is a good start @david4096, and contains a lot of the stuff we'll need to make progress on this. However, there's nothing really complete as such here. How do you want to play this? Merge this much first and create a bunch of tickets to address specific problems or try to clean things up a bit more here in this PR?

**Comment by david4096** [comment]
Yes, let's merge this and knock out those issues in separate PRs. @jeromekelleher @bwalsh

**Comment by bwalsh** [comment]
@david4096  this looks good.  yesterday we talked about me doing the merge - however it doesn't look like I have authorization ?

**Pull request merged**